packages/devtools_app/lib/src/shared/utils/utils.dart

```diff
@@ -185,17 +185,39 @@ extension IsKeyType on KeyEvent {
   bool get isKeyDownOrRepeat => this is KeyDownEvent || this is KeyRepeatEvent;
 }
 
+/// A helper class for [Timer] functionality, where the callbacks are debounced.
+class Debouncer {
+  Debouncer({required this.duration});
+
+  final Duration duration;
+  Timer? _activeTimer;
+
+  /// Invokes the [callback] once after the specified [duration] has elapsed.
+  ///
+  /// If multiple invokations are called while the timer is running, the
+  /// callback will be further delayed.
+  void run(void Function() callback) {
```

Comment:
exit early if dispose has been called. Make this class extend Disposable to get access to the disposed getter

Reply:
Done!